File: FinanceApp.Shared/Dto/Income/UpdateIncome.cs
using FinanceApp.Shared.Dto.Base;

namespace FinanceApp.Shared.Dto.Income
{
    public class UpdateIncome : UpdateDto
    {
        public string Name { get; set; }
        public int Recurrence { get; set; }
        public double Amount { get; set; }
        public DateTime InitialDate { get; set; }
        public DateTime? EndDate { get; set; }
        public bool IsEndless { get; set; }
        public int? TimesRecurrence { get; set; }
    }
}

File: FlutterApp/finance_app/lib/models/income/update_income.dart
class UpdateIncome {
  String id;
  String name;
  int recurrence;
  double amount;
  DateTime initialDate;
  DateTime? endDate;
  bool isEndless;
  int timesRecurrence;

  UpdateIncome(
      {required this.id,
      required this.name,
      required this.recurrence,
      required this.amount,
      required this.initialDate,
      required this.endDate,
      required this.isEndless,
      required this.timesRecurrence});

  Map<String, dynamic> toJson() => {
        'id': id,
        'name': name,
        'recurrence': recurrence,
        'amount': amount,
        'initialDate': initialDate.toString(),
        'endDate': endDate?.toString(),
        'isEndless': isEndless,
        'timesRecurrence': timesRecurrence,
      };
}
